### ISSUE TITLE
Why's it going direct in balancers.strategy.type = leastPing 

### ISSUE THREAD

**Issue by BI7PRK** [XTLS/Xray-core#872]
## Version
Xray-core v1.5.2

## Configuration scene
***DNS servers:***  `...["119.29.29.29", "208.67.222.222" ]...` (tag=dns_servers)

***Routing:*** `dns_in --> dns_out_balancer --> dns_out_s1 & dns_out_s2`

***designated IPs going outbound proxy:***
```json
{
	"type": "field",
	"ip": ["8.8.8.8", "208.67.222.222"],
	"network": "udp",
	"inboundTag": ["dns_servers"],
	"outboundTag": "proxy"
}
```

## balancers configs and performance result
- balancers.strategy.type = random
```log
Wed Dec 29 01:44:01 2021 daemon.info xray[23510]: 2021/12/29 01:44:01 127.0.0.1:54559 accepted udp:208.67.222.222:53 [dns_out_s1]
Wed Dec 29 01:44:01 2021 daemon.info xray[23510]: 2021/12/29 01:44:01 127.0.0.1:56768 accepted udp:208.67.222.222:53 [dns_out_s2]
```
- balancers.strategy.type = leastPing  ***(Why's it going direct?)***
```log
Wed Dec 29 01:48:18 2021 daemon.info xray[23798]: 2021/12/29 01:48:18 127.0.0.1:38337 accepted udp:208.67.222.222:53 [direct]
Wed Dec 29 01:48:19 2021 daemon.info xray[23798]: 2021/12/29 01:48:19 127.0.0.1:33896 accepted udp:208.67.222.222:53 [direct]
```

**Comment by yuhan6665** [comment]
配置 “observatory” 了吗
参考 https://github.com/XTLS/Xray-core/issues/914

**Comment by BI7PRK** [comment]
是的，连接观测配置为：
```JSON
"observatory": {
		"probeURL": "https://api.github.com/_private/browser/stats",
		"probeInterval": "10m",
		"subjectSelector": [
			"vless_"
		]
	}
```

**Comment by yuhan6665** [comment]
Closed in #1146
